crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -167,6 +168,31 @@ pub struct UploadOperationData {
     pub operation: CumulocitySupportedOperations,
 }
 
+/// Represents a pending download performed by the downloader from the FTS.
+///
+/// Functions which download files from the tedge File Transfer Service as part of handling
+/// operations (e.g. when performing `log_upload` or `config_snapshot`, the relevant file is
+/// uploaded into FTS) will use this type for communicating with the Downloader actor.
+pub struct FtsDownloadOperationData {
+    pub download_type: FtsDownloadOperationType,
+    pub url: String,
+
+    // used to automatically remove the temporary file after operation is finished
+    pub destination_dir: tempfile::TempDir,
+
+    pub smartrest_topic: Topic,
+    pub clear_cmd_topic: Topic,
+    pub event_response_id: String,
+}
+
+/// Used to denote as type of what operation was the file downloaded from the FTS.
+///
+/// Used to dispatch download result to the correct operation handler.
+pub enum FtsDownloadOperationType {
```

Comment:
You could just reuse `CumulocitySupportedOperations` instead of this new struct.

Reply:
By using a different enum, I wanted to underline that these are the only operations in which we need to download from the FTS. By having a new enum with only these 2 variants, I don't have to do a catch-all `_ => {}` in matches, so that when a new variant is possibly added, the compiler will automatically highlight all the matches where this variant is not handled.